adapter-base-spring/src/main/java/org/eclipse/hono/adapter/spring/AbstractMessagingClientConfig.java

```diff
@@ -0,0 +1,255 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.adapter.spring;
+
+import java.util.Optional;
+
+import org.eclipse.hono.adapter.AdapterConfigurationSupport;
+import org.eclipse.hono.adapter.MessagingClientSet;
+import org.eclipse.hono.adapter.MessagingClients;
+import org.eclipse.hono.adapter.client.command.CommandResponseSender;
+import org.eclipse.hono.adapter.client.command.amqp.ProtonBasedCommandResponseSender;
+import org.eclipse.hono.adapter.client.command.kafka.KafkaBasedCommandResponseSender;
+import org.eclipse.hono.adapter.client.telemetry.EventSender;
+import org.eclipse.hono.adapter.client.telemetry.TelemetrySender;
+import org.eclipse.hono.adapter.client.telemetry.amqp.ProtonBasedDownstreamSender;
+import org.eclipse.hono.adapter.client.telemetry.kafka.KafkaBasedEventSender;
+import org.eclipse.hono.adapter.client.telemetry.kafka.KafkaBasedTelemetrySender;
+import org.eclipse.hono.client.HonoConnection;
+import org.eclipse.hono.client.SendMessageSampler;
+import org.eclipse.hono.client.kafka.KafkaProducerConfigProperties;
+import org.eclipse.hono.client.kafka.KafkaProducerFactory;
+import org.eclipse.hono.client.kafka.consumer.KafkaConsumerConfigProperties;
+import org.eclipse.hono.config.ClientConfigProperties;
+import org.eclipse.hono.config.ProtocolAdapterProperties;
+import org.eclipse.hono.util.CommandConstants;
+import org.eclipse.hono.util.Constants;
+import org.eclipse.hono.util.MessagingType;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Scope;
+
+import io.opentracing.Tracer;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+
+
+/**
+ * A base class that provides helper methods for configuring messaging clients.
+ */
+@Configuration
+public abstract class AbstractMessagingClientConfig extends AdapterConfigurationSupport {
```

Comment:
The naming sounds a bit strange to me. My understanding is that `AbstractMessagingClientConfig` should not only be used in adapters but also in other services and therefore should not extend a class with "Adapter" in the name. Accordingly, the name of the` AdapterConfigurationSupport#getAdapterName()` method does not fit anymore.

Reply:
Yes, I also thought about changing this but was reluctant as this will add more changes to this PR. Hence I wanted to wait upon other opinions on that and change it in a separate PR.

It is also strange to me that it is an abstract class an not an interface...